### ISSUE TITLE
Tests and docs

### ISSUE THREAD

**Issue by KhraksMamtsov** [incetarik/fp-ts-stream#2]
Thank you for this library!
Could you please add tests and docs for it?)

**Comment by incetarik** [comment]
You are welcome!

The code is documented and as far as I know, there is no place undocumented, even returned partial functions are documented but if you are saying a website of the documentation of the library, I have not done it yet and I'm busy these days so I can't give you a certain date for it.

I tried to keep the functions the same with the array functions as much as possible, so what you know from there is valid here. `AsyncStream` might be an exception but you have very few things different in terms of functions and their names and the ways they work.

Well, for tests, you are right, adding tests would be beautiful but I did not do it for two things:
1. I find the functional programming provides types that are explaining themselves and usually very easy to understand (as they are usually short) and until now I have not seen any error and nobody came to me with an error to fix.
2. I also usually find most of the functional libraries have no tests, which I believe, due to the very same reason I have mentioned. Apart from that, adding tests would increase the package size.
